data/moves.js

```diff
@@ -10674,8 +10674,14 @@ exports.BattleMovedex = {
 					// Run through each decision in queue to check if the Pursuit user is supposed to mega evolve this turn.
 					// If it is, then mega evolve before moving.
 					var willMegaEvo = false;
-					for (var j = 0; j < this.queue.length; j++) {
-						if (this.queue[j].pokemon === sources[i] && this.queue[j].choice === 'megaEvo') willMegaEvo = true;
+					if (sources[i].canMegaEvo) {
+						for (var j = 0; j < this.queue.length; j++) {
+							if (this.queue[j].pokemon === sources[i] && this.queue[j].choice === 'megaEvo') {
+								this.queue.splice(j, 1);
+								willMegaEvo = true;
+								break;
+							}
+						}
 					}
 					if (willMegaEvo) this.runMegaEvo(sources[i]);
```

Comment:
We can just replace `willMegaEvo = true;` with `this.runMegaEvo(sources[i]);` and save ourselves a flag variable.